0_overview.md

```diff
@@ -66,3 +66,63 @@ Cowboy is an HTTP server written in Erlang by Loïc Hoguin of [99s](http://ninen
+
+- Erlang
+Elixir source code compiles to Erlang byte code which runs on the Erlang Virtual Machine. That means we must have Erlang installed on our system - in addition to Elixir - in order to work with Phoenix. The Elixir site also has [Erlang installation instructions](http://elixir-lang.org/install.html#installing-erlang).
+
+- node.js
+Node is a soft dependency. We don't absolutely need it, but without it, we must take care when we generate new Phoenix applications.
+
+By default, Phoenix will use brunch.io for asset compilation. Brunch.io uses the node package manager (npm) to install its dependencies, and npm requires node.js.
+
+If we choose not to use brunch.io, we can simply use the `--no-brunch` flag when generating our application. At that point, we would no longer need node.js, or npm. We would, however, have to provide another solution for asset compilation.
+
+Without the `--no-brunch` flag, when we generate a new Phoenix app, Phoenix will ask if we want to install brunch.io dependencies.
+
+If we have node.js on our system, we can choose to install the brunch.io dependencies right away, or later by running `npm install`. Neither option will cause problems.
+
+If we don't have node.js installed on our system, but we say yes to installing brunch.io dependencies anyway, this is where issues can creep in. Nothing will immediately fail, but when we try to start our application, we will get an error.
+
+```console
+[error] Could not start watcher "node", executable does not exist
+```
+
+Phoenix does its best to help us out here. This is the actual prompt we see wile generating a new application if we don't have node.js installed.
+
+```console
+Phoenix uses an optional build tool called brunch.io that
+requires npm but you don't have npm in your system. Would
+you still like to copy brunch.io files? [Yn]
+```
+
+If we say yes to this, Phoenix gives us a warning. It lets us know where to get node.js installation instructions, and it tells us how to install our brunch dependencies once we have node.js installed.
+
+```console
+Brunch was setup for static assets, but dependencies were
+not installed via npm. Installation instructions for node.js,
+which includes npm, can be found at http://nodejs.org.
+
+Install your brunch dependencies by running inside your app:
+
+$ npm install
+```
+
+When installing node.js, it is important to install a recent stable version. Currently, that means a version in the 0.12.x range. Some users have reported problems with significantly older versions of node - versions in the 0.10.x range, for example.
+
+- PostgreSQL
+PostgreSQL is a relational database server. When we work with Ecto models in these guides, we will use PostgreSQL and the Postgrex adapter for it. In order to follow along with the examples, we will need to install PostgreSQL on our system. The PostgreSQL wiki has [installation guides](https://wiki.postgresql.org/wiki/Detailed_installation_guides) for a number of different systems.
```

Comment:
We can mention one can use mysql by passing `--database mysql` when creating a new phoenix application.